### ISSUE TITLE
Viewport transX jumping when zooming Y scale (dynamic/realtime data)

### ISSUE THREAD

**Issue by kzhaowmm** [danielgindi/Charts#2719]
I've noticed that the viewport's transX (position on the X) jumps around when pinching the zoom on the Y scale. 
I also notice that the amount that the data jumps seems dependent on the left axis maximum. Setting this to smaller amounts tends to make the jump less.

Lastly, note that I have real time data coming that calls notifyDataSetChanged() and setNeedsDisplay() roughly once per second.

Is there any way to fix the transX to not move when zooming on the Y scale? Currently the x axis jumps back a few hundred data points when pinching to zoom in, and then jumps back to the original state when zooming back out.

**Comment by liuxuan30** [comment]
are you able to debug on your side? We could use your help

**Comment by kzhaowmm** [comment]
I figured it out: the trans X was jumping when the viewport offset rescaling based on the Y axis label width.

For example, if the Y label max jumped from 100 down to 90, the extra spacing on the left (3 characters down to 2 characters in label length) caused the viewport's left offset to be automatically rescaled. This rescaling causes a glitch in the transX when the new chart rerenders with extra pixels to the left.

I fixed this by manually setting the offset of the viewport, overriding any automatic resizing. 

chart?.setViewPortOffsets(**left: 25**, top: 20, right: 20, bottom: 25)
